main.js

```diff
@@ -1,32 +1,140 @@
+                let {top:topBounceDiv,left:leftBounceDiv,bottom:bottomBounceDiv,right:rightBounceDiv} = bounceDiv.getBoundingClientRect();
+        
+                let diffRightLeft = rightBounceDiv-leftBounceDiv
+                let difTopBottom = bottomBounceDiv - topBounceDiv
+        
+                for(let i=0;i<ballsNumber;i++){
 
-});
+                    let newDiv = createCircle()
+                    let xCoord = (leftBounceDiv+randomIntFromInterval(0,diffRightLeft-parseInt(newDiv.style.width)))
+                    let yCoord = (topBounceDiv+randomIntFromInterval(0,difTopBottom-parseInt(newDiv.style.width)))
+                    
+                    newDiv.style.left=xCoord+"px"
+                    newDiv.style.top=yCoord+"px"
+        
+                    circlesElements.push({
+                        dom:newDiv,
+                        x:xCoord,
+                        y:yCoord,
+                        step:randomIntFromInterval(1,10),
+                        direction:randomIntFromInterval(0,1),
+                        slope:Math.random()
+                    })
+                        bounceDiv.append(newDiv)
+                }
+                
+        
+                bounceInterval = setInterval(()=>{
+                    console.log("Bounce interval")
+                    let {top:topBounceDiv,left:leftBounceDiv,bottom:bottomBounceDiv,right:rightBounceDiv} = bounceDiv.getBoundingClientRect();
+                    circlesElements.forEach((circle)=>{
+        
+                        let circleRadius = parseInt(circle.dom.style.width)
+                        let direction = circle.direction
+                        if(direction == 0){
+                            circle.x += circle.step
+                        } else {
+                            circle.x -= circle.step
+                        }
+        
+        
+                        circle.y += circle.slope*circle.step
+        
+                        if(circle.x > rightBounceDiv - circleRadius){ // Right
+                            circle.direction = (direction+1)%2
+                            circle.x = rightBounceDiv - circleRadius
+                        } else if (circle.x < leftBounceDiv){
+                            circle.direction = (direction+1)%2
+                            circle.x = leftBounceDiv
+                        }
+        
+                        if(circle.y > bottomBounceDiv - circleRadius){ // Right
+                            circle.slope *= -1;
+                            circle.y = bottomBounceDiv - circleRadius
+                            
+                        }  else if(circle.y < topBounceDiv){
+                            circle.slope *= -1;
+                            circle.y =  topBounceDiv
+                        }
+        
+                        circle.dom.style.left = circle.x+"px"
+                        circle.dom.style.top = circle.y+"px"
+                    })
+                },15)
```

Comment:
I tested multiple values and I chose **15** because it had the best results. Should I try another value or find the correct value through another process?

Reply:
Well, I would think about the number of frames per sec we want to achieve, so the animation appears smooth. Usually, [60fps ](https://developer.mozilla.org/en-US/docs/Tools/Performance/Frame_rate)should be enough.

Reply:
Got it. Changed it to 1000/FPS = 16.6